File: fibonacci/iterative.py
import utils

def fibIter(n):
    if n < 2:
        return n
    fibPrev = 1
    fib = 1
    for num in range(2, n):
        fibPrev, fib = fib, fib + fibPrev
    return fib

@utils.bench
def run(n):
    return fibIter(n)

n=25000
s = utils.limit("{}".format(run(n)), 10)
print('n={} fib={}'.format(n, s))

File: fibonacci/iterative.js
const utils = require('./utils');


const fibIter = (n) => {
  if (n < 2) {
    return n;
  }
  let a = 0n;
  let b = 1n;
  for (let num = n - 1; num > 0; num--) {
    a = a + b;
    [a, b] = [b, a];
  }
  return b;
}

const run = (n) => utils.bench(fibIter)(n);


n = 80000;
s = utils.limit(`${run(n)}`, 10);
console.log(`n=${n} fib=${s}`);
